task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -48,11 +52,10 @@ class KubernetesPodExecutor(TaskExecutor):
     def __init__(self, namespace: str) -> None:
         self.kube_client = KubeClient()
         self.namespace = namespace
-
         self.task_metadata: PMap[str, KubernetesTaskMetadata] = pmap()
         self._lock = threading.RLock()
 
-    def run(self, task_config: KubernetesTaskConfig) -> None:
+    def run(self, task_config: KubernetesTaskConfig) -> str:
```

Comment:
we probably want to update the return type if we're gonna return the task name on success and None otherwise
```suggestion
    def run(self, task_config: KubernetesTaskConfig) -> Optional[str]:
```